unblob/handlers/archive/cpio.py

```diff
@@ -92,20 +179,56 @@ def calculate_chunk(  # noqa: C901
             # heuristics 3: check mode field
             is_valid = file_type in C_FILE_TYPES and sticky_bit in C_STICKY_BITS
             if not is_valid:
-                return None
-
-            file_with_offset.seek(c_filesize, io.SEEK_CUR)
-            current_offset += self._pad_content(header, c_filesize, c_namesize)
-
-        end_offset = start_offset + self._pad_file(
-            file_with_offset, current_offset - start_offset
-        )
-        if start_offset == end_offset:
-            return None
-        return ValidChunk(
-            start_offset=start_offset,
-            end_offset=end_offset,
+                raise InvalidInputFormat("CPIO entry mode is invalid.")
+
+            self.entries.append(
+                CPIOEntry(
+                    start_offset=current_offset,
+                    size=c_filesize,
+                    dev=self._calculate_dev(header),
+                    mode=c_mode,
+                    rdev=self._calculate_rdev(header),
+                    path=Path(filename),
+                )
+            )
+
+            self.offset_file.seek(c_filesize, io.SEEK_CUR)
+            current_offset += self._pad_content(c_filesize)
+
+        self.end_offset = self.start_offset + self._pad_file(
+            self.offset_file, current_offset - self.start_offset
         )
+        if self.start_offset == self.end_offset:
+            raise InvalidInputFormat("Invalid CPIO archive.")
+
+    def dump_entries(self, fs: FileSystem):
+        for entry in self.entries:
+            # skip entries with "." as filename
+            if not entry.path.name:
+                continue
```

Comment:
I was surprised by this fragment. I would have expected `.` as name for `Path(".")` as `..` is returned for `Path("..")`.
This behavior is not documented at https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.name, so it might be a bug (https://bugs.python.org/issue37130), that is fixed later.
I propose the below change, which will work with both the current bad behavior, and with the fixed `Path.name` behavior:

```suggestion
            # skip entries with "." as filename, currently Path(".").name is ""
            #  see https://bugs.python.org/issue37130
            if entry.path.name in ("", "."):
                continue
```

Reply:
Yeah it's the most counter-intuitive part of pathlib to me. I implemented your suggestion.